librosa/core/constantq.py

```diff
@@ -728,10 +730,257 @@ def icqt(C, sr=22050, hop_length=512, fmin=None, bins_per_octave=12,
     return y
 
 
+@cache(level=20)
+def vqt(y, sr=22050, hop_length=512, fmin=None, n_bins=84, gamma=None,
+        bins_per_octave=12, tuning=0.0, filter_scale=1,
+        norm=1, sparsity=0.01, window='hann',
+        scale=True, pad_mode='reflect', res_type=None):
+    '''Compute the variable-Q transform of an audio signal.
+
+    This implementation is based on the recursive sub-sampling method
+    described by [1]_.
+
+    .. [1] Schörkhuber, Christian, Anssi Klapuri, Nicki Holighaus, and Monika Dörfler.
+        "A Matlab toolbox for efficient perfect reconstruction time-frequency transforms with log-frequency resolution."
+        In Audio Engineering Society Conference: 53rd International Conference: Semantic Audio.
+        Audio Engineering Society, 2014.
+
+    Parameters
+    ----------
+    y : np.ndarray [shape=(n,)]
+        audio time series
+
+    sr : number > 0 [scalar]
+        sampling rate of `y`
+
+    hop_length : int > 0 [scalar]
+        number of samples between successive CQT columns.
+
+    fmin : float > 0 [scalar]
+        Minimum frequency. Defaults to C1 ~= 32.70 Hz
+
+    n_bins : int > 0 [scalar]
+        Number of frequency bins, starting at `fmin`
+
+    gamma : number > 0 [scalar]
+        Bandwidth offset for determining filter lengths.
```

Comment:
I would love to see a more thorough documentation of the role of gamma. 

* What is the unit of `gamma`?
* What happens in the default case `gamma=None`? 
* What happens for very large `gamma`? 
* What are some typical values? (from the Schörkhuber group)

Reply:
Although it was not explicitly stated, I think we can infer that the units of gamma are hertz, since it is added as a constant offset to bandwidth.

I don't think it makes sense to default gamma to None. It should instead be defaulted to zero, where it would just give us the standard CQT. Similarly, with regard to the question Brian posed earlier, my opinion is that the CQT and VQT should be combined into one function, if the octave-wise iterative processing does not require significant changes (I need to read the paper again to better understand how they account for gamma in their implementation and reconstruction).

A very large gamma is not advised in my opinion. In the paper, I don't recall them going past gamma=25, but even that gives very poor frequency resolution at lower frequencies. However, it should be noted once again that they were using an alternate convention for calculating bandwidth, and my assessment of poor frequency resolution at lower frequencies for gamma=25 was made while using the convention currently adopted by librosa.

I think the range [0, 5] is where I would usually place the parameter for practical scenarios. Besides gamma=0 giving the CQT, there is one other special case where gamma can be set to make bandwidths proportional, not to be confused with equal, to the ERB scale. I think special case gamma would be somewhere in this range, but depends on the number of bins per octave.

Reply:
>     * What happens in the default case `gamma=None`?

Nothing yet, but the idea was to automatically fill a value based on ERB.  I don't think there's much sense in having 0 be the default if the two functions remain separate (vqt and cqt), but of course, if we merge them, then `gamma=0` is a totally fine (and expected) default.

Otherwise, I agree with @cwitkowitz's comments.

Reply:
True, if the functions remain separate, the ERB setting would be a good default for gamma='None'.